### PR TITLE
feat: new dashboard layout

### PR DESCRIPTION
**Pull request by gregogun**
This pr adds the layout for the new dashboard. 

To inspect, navigate to `/dashboard` (will be updated to home route when remaining UI and functionality is added and validated).

### CODE REVIEW

File: sigle/src/modules/layout/components/NewDashboardLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+interface NavItemProps {
+  href?: string;
+  children: React.ReactNode;
+}
+
+const NavItem = ({ children, ...props }: NavItemProps) => {
+  const router = useRouter();
+  return (
+    <Box
```
[the remark this answers is not in the document]
Comment (gregogun) [reply sigle/src/modules/layout/components/NewDashboardLayout.tsx:31]: ah yes it is, thanks for spotting that!

File: sigle/src/modules/layout/components/NewDashboardLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      {children}
+    </Box>
+  );
+};
+
+interface NewDashboardLayoutProps {
+  children: React.ReactNode;
+}
+
+export const NewDashboardLayout = ({ children }: NewDashboardLayoutProps) => {
```
Comment (pradel) [inline sigle/src/modules/layout/components/NewDashboardLayout.tsx:61]: for this one let's just call it DashboardLayout as the old one is deleted :)

File: sigle/src/modules/layout/components/NewDashboardLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <FullScreen>
+      <AppHeader />
+      <Container
+        css={{
+          display: 'flex',
+          gap: 103,
+          mt: '$10',
+          width: '100%',
+          height: '100%',
```
Comment (pradel) [inline sigle/src/modules/layout/components/NewDashboardLayout.tsx:71]: this can probably be removed as there is some content now

**Review by pradel — CHANGES_REQUESTED** [review]
Some visual issues I had while testing:
1. Logo is cropped in the header
Screenshot 2022-03-22 at 17 39 15

2. Wdyt about removing the old buttons, both actions are already in the new header
Screenshot 2022-03-22 at 17 39 27

3. When there is a lot of content, the content is passing on top of the footer 
Screenshot 2022-03-22 at 17 42 28

Screenshot 2022-03-22 at 17 39 42

File: sigle/src/modules/layout/components/DashboardHeader.tsx [hunk 4]
```diff
@@ -131,19 +131,6 @@ export const DashboardPageTitle = ({ title }: DashboardPageTitleProps) => {
...
-            size="large"
-            as="a"
-            href={`/${user?.username}`}
-            target="_blank"
-          >
-            Visit my blog{' '}
-            <EyeOpenIcon width={18} height={18} style={{ marginLeft: 8 }} />
-          </VisitButton>
-          {!loadingCreate && (
-            <Button onClick={handleCreateNewPrivateStory}>New story</Button>
```
Comment (pradel) [inline sigle/src/modules/layout/components/DashboardHeader.tsx:144]: can we delete the handleCreateNewPrivateStory function in this file?

File: sigle/src/modules/layout/components/NewDashboardLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+export const DashboardLayout = ({ children }: DashboardLayoutProps) => {
+  return (
+    <FullScreen>
+      <AppHeader />
+      <Container
+        css={{
+          display: 'grid',
+          gridTemplateColumns: '1fr 820px 1fr',
```
[the remark this answers is not in the document]
Comment (gregogun) [reply sigle/src/modules/layout/components/NewDashboardLayout.tsx:68]: Quentin confirmed that width for both is 826px so will update the config :)

File: sigle/src/modules/layout/components/DashboardLayout.tsx [hunk 6]
```diff
@@ -54,11 +60,62 @@ const NavItem = ({ children, ...props }: NavItemProps) => {
   );
 };
 
+interface AccordionNavItemProps {
+  href?: string;
+  children: React.ReactNode;
+}
+
```
Comment (gregogun) [inline sigle/src/modules/layout/components/DashboardLayout.tsx:67]: created a seperate component from `NavItem` as there were a few differences between the nav items in the sidebar and the ones in the accordion

File: sigle/src/modules/layout/components/AppHeader.tsx [hunk 7]
```diff
@@ -4,14 +4,26 @@ import {
   TwitterLogoIcon,
   DiscordLogoIcon,
   SunIcon,
+  HamburgerMenuIcon,
```
Comment (pradel) [inline sigle/src/modules/layout/components/AppHeader.tsx:7]: regarding the mobile menu of this one, will it be impacted by the coming changes that we will have with the top right corner? It seems to me that these changes should be made in a separate pull request.

**Review by pradel — CHANGES_REQUESTED** [review]
Good job, new layout is 🔥 

- Can you please create another changeset for the tailwind package changes?
- You have some git conflicts on this branch
- The mobile navigation menu should be shown just on the mobile right? 😅 
- Can you check why this warning is happening?
Screenshot 2022-03-30 at 11 41 33

File: sigle/src/modules/layout/components/DashboardLayout.tsx (hunk 6 above)
Comment (gregogun) [reply sigle/src/modules/layout/components/DashboardLayout.tsx:67]: This approach makes much more sense, and also solves the warning in the console, thanks! 

Following the design, the items still need background colour when matching that particular path. Do you think it's better to add this functionality to them individually or create a wrapping component (similar to above) where that logic can be added? Hope that makes sense 😅

Comment (pradel) [reply sigle/src/modules/layout/components/DashboardLayout.tsx:67]: I think using variants is the way for this :)

File: sigle/src/modules/layout/components/DashboardLayout.tsx [hunk 8]
```diff
@@ -1,257 +1,193 @@
...
+            '@md': {
+              maxWidth: 724,
+            },
+
+            '@lg': {
+              maxWidth: 834,
+            },
+
+            '@xl': {
+              maxWidth: 826,
```
[the remark this answers is not in the document]
Comment (gregogun) [reply sigle/src/modules/layout/components/DashboardLayout.tsx:166]: checked with Quentin and he confirmed that this is intended but should automatically happen with the padding so will remove the hardcoded values :)

File: sigle/src/modules/layout/components/DashboardLayout.tsx [hunk 9]
```diff
@@ -1,257 +1,182 @@
...
+            <NavItem selected={router.pathname === '/settings'}>
+              Settings
+            </NavItem>
+          </Link>
+        </Sidebar>
+        <Box
+          css={{
+            mb: '$5',
+            mx: 'auto',
+            width: '100%',
```
Comment (pradel) [inline sigle/src/modules/layout/components/DashboardLayout.tsx:140]: These 2 props can probably be removed?

File: sigle/src/modules/layout/components/DashboardLayout.tsx [hunk 10]
```diff
@@ -1,257 +1,182 @@
...
+            collapsible
+            type="single"
+          >
+            <AccordionItem value="item1">
+              <AccordionTrigger>{triggerName}</AccordionTrigger>
+              <AccordionContent>
+                <Link href="/" passHref>
+                  <NavItem hidden={router.pathname === '/'} variant="accordion">
+                    Drafts
+                  </NavItem>
```
Comment (pradel) [inline sigle/src/modules/layout/components/DashboardLayout.tsx:154]: If the goal is to hide the element would doing smth like this in react make sense?
```
                {router.pathname !== '/' ? <Link href="/" passHref>
                  <NavItem variant="accordion">
                    Drafts
                  </NavItem> : null}
```

File: sigle/src/modules/layout/components/AppHeader.tsx [hunk 11]
```diff
@@ -42,8 +42,18 @@ const Header = styled('header', Container, {
   width: '100%',
 
   '@md': {
+    px: 0,
```
Comment (pradel) [inline sigle/src/modules/layout/components/AppHeader.tsx:45]: Could you please explain the reason of this change?

File: sigle/src/modules/layout/components/AppHeader.tsx [hunk 12]
```diff
@@ -57,7 +67,11 @@ const StatusDot = styled('div', {
   borderRadius: '$round',
 });
 
-export const AppHeader = () => {
+interface AppHeaderProps {
+  children?: React.ReactNode;
```
Comment (pradel) [inline sigle/src/modules/layout/components/AppHeader.tsx:71]: This prop is not used right?

File: sigle/src/modules/layout/components/AppHeader.tsx [hunk 13]
```diff
@@ -114,127 +128,136 @@ export const AppHeader = () => {
...
-              logout
-              <HoverCardArrow />
-            </HoverCardContent>
-          </HoverCard>
-        ) : (
-          <Flex gap="6">
-            <IconButton
-              as="a"
-              href={sigleConfig.twitterUrl}
+    <>
```
Comment (pradel) [inline sigle/src/modules/layout/components/AppHeader.tsx:131]: Looks like fragment only has one element, can you please remove it?

**Review by pradel — COMMENTED** [review]
Last small nits :)

File: sigle/src/modules/layout/components/DashboardLayout.tsx (hunk 10 above)
Comment (gregogun) [reply sigle/src/modules/layout/components/DashboardLayout.tsx:154]: Ah yep that makes more sense 😅

File: sigle/src/modules/layout/components/AppHeader.tsx (hunk 11 above)
Comment (gregogun) [reply sigle/src/modules/layout/components/AppHeader.tsx:45]: I think this was from where I was setting the maxWidth at each media query, but didn't take into account the padding that is already there as part of the container. Will remove this and fixed maxWidth sizes 👍

File: sigle/src/modules/layout/components/AppHeader.tsx (hunk 12 above)
Comment (gregogun) [reply sigle/src/modules/layout/components/AppHeader.tsx:71]: Sure, will remove

File: sigle/src/stitches.config.ts [hunk 14]
```diff
@@ -61,6 +61,14 @@ const orangeDark = {
   orange12: '#FF802B',
 };
 
+const red = {
```
Comment (pradel) [inline sigle/src/stitches.config.ts:64]: Are these ones used?

**Review by pradel — CHANGES_REQUESTED** [review]
Last comment and let's merge it tomorrow 🚀

**Review by pradel — APPROVED** [review]
